int64 x
int64 y
int64 depth
int64 image_width
